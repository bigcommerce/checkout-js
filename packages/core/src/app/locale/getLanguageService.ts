import { createLanguageService, LanguageConfig, LanguageService } from '@bigcommerce/checkout-sdk';

import { FALLBACK_LOCALE, FALLBACK_TRANSLATIONS } from './translations';

let languageService: LanguageService | undefined;

export default function getLanguageService(): LanguageService {
    languageService =
        languageService ??
        createLanguageService({
            fallbackLocale: FALLBACK_LOCALE,
            fallbackTranslations: FALLBACK_TRANSLATIONS,
        });

    return languageService;
}

export type InitializeLanguageService = typeof initializeLanguageService;

export function initializeLanguageService(config: LanguageConfig): LanguageService {
    languageService = createLanguageService({
        ...config,
        defaultLocale: config.locale,
        defaultTranslations: config.defaultTranslations,
        fallbackLocale: FALLBACK_LOCALE,
        fallbackTranslations: FALLBACK_TRANSLATIONS,
    });

    return languageService;
}
