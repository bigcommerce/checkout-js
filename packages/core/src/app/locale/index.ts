export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
