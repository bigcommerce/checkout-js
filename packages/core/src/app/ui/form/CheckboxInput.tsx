import classNames from 'classnames';
import React, { forwardRef, ReactNode, Ref } from 'react';

import Input, { InputProps } from './Input';
import Label from './Label';

export interface CheckboxInputProps extends InputProps {
    additionalClassName?: string;
    label: ReactNode;
    value: string;
    checked: boolean;
    themeV2?: boolean;
}

const CheckboxInput = forwardRef(
    (
        { additionalClassName, label, id, testId, themeV2 = false, ...rest }: CheckboxInputProps,
        ref: Ref<HTMLInputElement>,
    ) => (
        <>
            <Input
                {...rest}
                className={classNames(
                    'form-checkbox',
                    'optimizedCheckout-form-checkbox',
                    additionalClassName,
                )}
                id={id}
                ref={ref}
                testId={testId}
                type="checkbox"
            />
            <Label additionalClassName={themeV2 ? 'body-regular' : ''} htmlFor={id}>{label}</Label>
        </>
    ),
);

export default CheckboxInput;
