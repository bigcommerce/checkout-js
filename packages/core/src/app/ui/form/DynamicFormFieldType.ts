enum DynamicFormFieldType {
    telephone = 'tel',
    dropdown = 'dropdown',
    number = 'number',
    password = 'password',
    checkbox = 'checkbox',
    multiline = 'multiline',
    date = 'date',
    radio = 'radio',
    text = 'text',
}

export default DynamicFormFieldType;
