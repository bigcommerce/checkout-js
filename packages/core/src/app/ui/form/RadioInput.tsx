import classNames from 'classnames';
import React, { forwardRef, ReactNode, Ref } from 'react';

import Input, { InputProps } from './Input';
import Label from './Label';

export interface RadioInputProps extends InputProps {
    additionalClassName?: string;
    label: ReactNode;
    value: string;
    checked: boolean;
    newFontStyle?: boolean;
}

const RadioInput = forwardRef(
    (
        { additionalClassName, label, value, checked, id, newFontStyle = false, ...rest }: RadioInputProps,
        ref: Ref<HTMLInputElement>,
    ) => (
        <>
            <Input
                {...rest}
                checked={checked}
                className={classNames(
                    'form-radio',
                    'optimizedCheckout-form-radio',
                    additionalClassName,
                )}
                id={id}
                ref={ref}
                type="radio"
                value={value}
            />
            <Label additionalClassName={newFontStyle ? 'body-regular' : ''} htmlFor={id}>{label}</Label>
        </>
    ),
);

export default RadioInput;
