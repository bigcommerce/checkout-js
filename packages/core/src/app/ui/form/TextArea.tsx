import classNames from 'classnames';
import React, { forwardRef, Ref, TextareaHTMLAttributes } from 'react';

export interface TextAreaProps extends TextareaHTMLAttributes<HTMLTextAreaElement> {
    additionalClassName?: string;
    testId?: string;
    useFloatingLabel?: boolean;
}

const TextArea = forwardRef(
    (
        { additionalClassName, testId, className, useFloatingLabel, ...rest }: TextAreaProps,
        ref: Ref<HTMLTextAreaElement>,
    ) => {
        return (
            <textarea
                {...rest}
                className={
                    className ||
                    classNames(
                        { 'floating-textarea': useFloatingLabel },
                        'form-input',
                        'optimizedCheckout-form-input',
                        additionalClassName,
                    )
                }
                data-test={testId}
                ref={ref}
            />
        );
    },
);

export default TextArea;
