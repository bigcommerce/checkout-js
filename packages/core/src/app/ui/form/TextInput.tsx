import classNames from 'classnames';
import React, { forwardRef, Ref } from 'react';

import Input, { InputProps } from './Input';

export interface TextInputProps extends InputProps {
    additionalClassName?: string;
    appearFocused?: boolean;
    type?: 'text' | 'password' | 'tel' | 'email' | 'number';
    useFloatingLabel?: boolean;
}

const TextInput = forwardRef(
    (
        {
            additionalClassName,
            useFloatingLabel,
            appearFocused,
            type = 'text',
            ...rest
        }: TextInputProps,
        ref: Ref<HTMLInputElement>,
    ) => (
        <Input
            {...rest}
            className={classNames(
                'form-input',
                'optimizedCheckout-form-input',
                { 'form-input--focus': appearFocused },
                { 'optimizedCheckout-form-input--focus': appearFocused },
                additionalClassName,
            )}
            ref={ref}
            type={type}
            useFloatingLabel={useFloatingLabel}
        />
    ),
);

export default TextInput;
