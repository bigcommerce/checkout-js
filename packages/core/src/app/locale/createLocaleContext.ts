import { createCurrencyService, StoreConfig } from '@bigcommerce/checkout-sdk';

import getLanguageService from './getLanguageService';
import { LocaleContextType } from './LocaleContext';

export default function createLocaleContext(config: StoreConfig): Required<LocaleContextType> {
    if (!config) {
        throw new Error('Missing configuration data');
    }

    const { inputDateFormat } = config;

    return {
        currency: createCurrencyService(config),
        date: {
            inputFormat: inputDateFormat,
        },
        language: getLanguageService(),
    };
}
