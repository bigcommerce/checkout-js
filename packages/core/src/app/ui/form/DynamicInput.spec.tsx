import { mount } from 'enzyme';
import React, { FunctionComponent } from 'react';
import ReactDatePicker from 'react-datepicker';

import { getStoreConfig } from '../../config/config.mock';
import { createLocaleContext, LocaleContext, LocaleContextType, WithDateProps } from '../../locale';

import CheckboxInput from './CheckboxInput';
import DynamicFormFieldType from './DynamicFormFieldType';
import DynamicInput, { DynamicInputProps } from './DynamicInput';
import RadioInput from './RadioInput';
import TextArea from './TextArea';
import TextInput from './TextInput';

describe('DynamicInput', () => {
    let localeContext: Required<LocaleContextType>;
    let date: Required<LocaleContextType>['date'];
    let DynamicInputTest: FunctionComponent<DynamicInputProps & WithDateProps>;

    beforeEach(() => {
        localeContext = createLocaleContext(getStoreConfig());
        date = localeContext.date;

        DynamicInputTest = (props) => (
            <LocaleContext.Provider value={localeContext}>
                <DynamicInput {...props} />
            </LocaleContext.Provider>
        );
    });

    it('renders text input for default input with passed props', () => {
        expect(
            mount(<DynamicInputTest date={date} id="field_33" name="x" />).html(),
        ).toMatchSnapshot();
    });

    it('renders textarea for multiline type', () => {
        expect(
            mount(
                <DynamicInputTest
                    date={date}
                    fieldType={DynamicFormFieldType.multiline}
                    id="field_33"
                    rows={4}
                />,
            )
                .find(TextArea)
                .prop('rows'),
        ).toBe(4);
    });

    it('renders date picker for date type', () => {
        const datePicker = mount(
            <DynamicInputTest
                date={date}
                fieldType={DynamicFormFieldType.date}
                id="field_33"
                max="2019-02-01"
                min="2019-01-01"
            />,
        );

        expect(datePicker.find(ReactDatePicker).props()).toMatchObject(
            expect.objectContaining({
                minDate: new Date('2019-01-01T00:00:00Z'),
                maxDate: new Date('2019-02-01T00:00:00Z'),
            }),
        );
    });

    it('renders checkbox input for checkbox type', () => {
        const component = mount(
            <DynamicInputTest
                date={date}
                fieldType={DynamicFormFieldType.checkbox}
                id="id"
                options={[
                    { value: 'x', label: 'X' },
                    { value: 'y', label: 'Y' },
                ]}
                value={['y']}
            />,
        );

        expect(component.find(CheckboxInput).at(0).props()).toEqual(
            expect.objectContaining({
                id: 'id-x',
                checked: false,
            }),
        );

        expect(component.find(CheckboxInput).at(1).props()).toEqual(
            expect.objectContaining({
                id: 'id-y',
                checked: true,
            }),
        );
    });

    it('renders radio type input for radio type', () => {
        const component = mount(
            <DynamicInputTest
                date={date}
                fieldType={DynamicFormFieldType.radio}
                id="id"
                onChange={jest.fn()}
                options={[
                    { value: 'x', label: 'X' },
                    { value: 'y', label: 'Y' },
                ]}
                value="y"
            />,
        );

        expect(component.find(RadioInput).at(0).props()).toEqual(
            expect.objectContaining({
                id: 'id-x',
                checked: false,
            }),
        );

        expect(component.find(RadioInput).at(1).props()).toEqual(
            expect.objectContaining({
                id: 'id-y',
                checked: true,
            }),
        );
    });

    it('renders number type input for number type', () => {
        expect(
            mount(
                <DynamicInputTest
                    date={date}
                    fieldType={DynamicFormFieldType.number}
                    id="field_33"
                />,
            )
                .find(TextInput)
                .prop('type'),
        ).toBe('number');
    });

    it('renders password type input for password type', () => {
        expect(
            mount(
                <DynamicInputTest
                    date={date}
                    fieldType={DynamicFormFieldType.password}
                    id="field_33"
                />,
            )
                .find(TextInput)
                .prop('type'),
        ).toBe('password');
    });

    it('renders tel type input for phone type', () => {
        expect(
            mount(
                <DynamicInputTest
                    date={date}
                    fieldType={DynamicFormFieldType.telephone}
                    id="field_33"
                />,
            )
                .find(TextInput)
                .prop('type'),
        ).toBe('tel');
    });

    it('renders select input with passed props', () => {
        expect(
            mount(
                <DynamicInputTest
                    date={date}
                    defaultValue="foo"
                    fieldType={DynamicFormFieldType.dropdown}
                    id="field_33"
                    name="select"
                    options={[
                        { label: 'Foo', value: 'foo' },
                        { label: 'Foo1', value: 'foo1' },
                    ]}
                    placeholder="Select an option"
                />,
            ).html(),
        ).toMatchSnapshot();
    });
});
