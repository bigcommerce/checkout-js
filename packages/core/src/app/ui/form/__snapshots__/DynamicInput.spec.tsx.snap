// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicInput renders select input with passed props 1`] = `"<div class=\\"dropdown-chevron\\"><div class=\\"icon\\"><svg height=\\"24\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></svg></div></div><select class=\\"form-select optimizedCheckout-form-select\\" data-test=\\"field_33-select\\" id=\\"field_33\\" name=\\"select\\"><option value=\\"\\">Select an option</option><option value=\\"foo\\" selected=\\"\\">Foo</option><option value=\\"foo1\\">Foo1</option></select>"`;

exports[`DynamicInput renders text input for default input with passed props 1`] = `"<input id=\\"field_33\\" type=\\"text\\" class=\\"form-input optimizedCheckout-form-input\\" name=\\"x\\" data-test=\\"field_33-text\\" value=\\"\\">"`;
